Tests/SentryTests/SentryCrash/SentryBinaryImageCacheTests.m

```diff
@@ -0,0 +1,227 @@
+#import "SentryCrashBinaryImageCache.h"
+#import "SentryCrashWrapper.h"
+#import <XCTest/XCTest.h>
+#include <mach-o/dyld.h>
+
+// Exposing test only functions from `SentryCrashBinaryImageCache.m`
+void sentry_setRegisterFuncForAddImage(void *addFunction);
+void sentry_setRegisterFuncForRemoveImage(void *removeFunction);
+void sentry_resetFuncForAddRemoveImage(void);
+
+static void (*addBinaryImage)(const struct mach_header *mh, intptr_t vmaddr_slide);
+static void (*removeBinaryImage)(const struct mach_header *mh, intptr_t vmaddr_slide);
+static NSMutableArray *mach_headers_test_cache;
+static NSMutableArray *mach_headers_initial_array;
+
+static void
+sentry_register_func_for_add_image(
+    void (*func)(const struct mach_header *mh, intptr_t vmaddr_slide))
+{
+    addBinaryImage = func;
+
+    if (mach_headers_initial_array) {
+        for (NSUInteger i = 0; i < mach_headers_initial_array.count; i++) {
+            NSValue *header = mach_headers_initial_array[i];
+            func(header.pointerValue, 0);
+        }
+    }
+}
+
+static void
+sentry_register_func_for_remove_image(
+    void (*func)(const struct mach_header *mh, intptr_t vmaddr_slide))
+{
+    removeBinaryImage = func;
+}
+
+static void
+cacheMachHeaders(const struct mach_header *mh, __unused intptr_t vmaddr_slide)
+{
+    [mach_headers_test_cache addObject:[NSValue valueWithPointer:mh]];
+}
+
+static void
+countNumberOfImagesInCache(__unused SentryCrashBinaryImage *image, void *context)
+{
+    int *counter = context;
+    (*counter)++;
+}
+
+static void
+addBinaryImageToArray(SentryCrashBinaryImage *image, void *context)
+{
+    NSMutableArray *array = (__bridge NSMutableArray *)context;
+    [array addObject:[NSValue valueWithPointer:image]];
+}
+
+@interface SentryBinaryImageCacheTests : XCTestCase
+
+@end
+
+@implementation SentryBinaryImageCacheTests
```

Comment:
`m`: As the `SentryBinaryImageCache` uses a mutex, it would be great to add some test cases checking if our locking strategy works as expected.